### PR TITLE
Allow to customize response error

### LINKED ISSUE #19: TooManyRequests error
The TooManyRequests error is returned as a text, which is not good for middleware in the API. Is it better to add a feature called json_response or make it a method like with_headers?

### PR DESCRIPTION
**Pull request by TheAwiteb**
This PR will fixes #19

### CODE REVIEW

**Comment by TheAwiteb** [comment]
it's missing a tests, i will add it, don't merge it

**Comment by TheAwiteb** [comment]
@AaronErhardt There is something incomplete in the tests, I did not know how to check it because it gives me the error and does not return a response.

https://github.com/AaronErhardt/actix-governor/blob/65dd1779e8d27545c0da959193e8221b76fe1bda/src/tests.rs#L560-L562
https://github.com/AaronErhardt/actix-governor/blob/65dd1779e8d27545c0da959193e8221b76fe1bda/src/tests.rs#L616-L621

File: src/key_extractor.rs [hunk 1]
```diff
@@ -17,6 +18,19 @@ pub trait KeyExtractor: Clone {
     /// Extraction method
     fn extract(&self, req: &ServiceRequest) -> Result<Self::Key, Self::KeyExtractionError>;
 
+    /// The content you want to show it when the rate limit is exceeded.<br>
+    /// The [`NotUntil`] will be passed to it and it has enough information.<br>
```
Comment (AaronErhardt) [inline src/key_extractor.rs]: Is there a reason why you use inline HTML? I personally don't like the `<br>` tags here.

File: src/lib.rs [hunk 2]
```diff
@@ -97,6 +97,16 @@
 //!
 //! Check out the [custom_key](https://github.com/AaronErhardt/actix-governor/blob/main/examples/custom_key.rs) example to see how a custom key extractor can be implemented.
 //!
+//! # Customize response error content
+//!
+//! By default, when the rate limit is exceeded the error will show up is `Too many requests, retry in {}s`<br>
+//! and content type it's plaintext, if you want to customize it you need to override the [`response_error_content`] function<br>
```
Comment (AaronErhardt) [inline src/lib.rs]: The same thing here. I don't thing the `<br>` makes sense here anyway.

Comment (AaronErhardt) [inline src/lib.rs]: My suggestion to improve readability:

 //! and the content type is plaintext. If you want to customize the message and content type you can override the [`response_error_content`] function.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good too, just a few minor things again :)

File: src/lib.rs (hunk 2 above)
Comment (TheAwiteb) [reply src/lib.rs]: I put it to write the line below it, if I didn't put it he would stifle it with the same line, and if I left a blank line the space would be as big as the distance between them and the example

> 
image

Comment (AaronErhardt) [reply src/lib.rs]: But that's the idiomatic way of writing Rust documentation / MarkDown. A paragraph should let the text flow naturally. Manual `<br>` tags only make things look odd on devices with small screens. On top of that, placing a manual `<br>` in the middle of a sentence doesn't make any sense either.

**Review by AaronErhardt — APPROVED** [review]
Looks good here too. Is it ready to merge?

**Comment by TheAwiteb** [comment]
> Is it ready to merge?

Yes ready 🎉

**Comment by AaronErhardt** [comment]
Thanks!

**Comment by TheAwiteb** [comment]
@AaronErhardt I think it's time for a new release 🚀

**Comment by AaronErhardt** [comment]
@TheAwiteb you probably know it better than me: did we have any breaking changes between 0.3 and now? At least the last two PRs shouldn't have any breaking changes so I'd bump the version to 0.3.1 then.

**Comment by TheAwiteb** [comment]
> did we have any breaking changes between 0.3 and now

No there is no BC, you can check the tests, there is no change in it

**Comment by TheAwiteb** [comment]
Hmmm, i will check

https://github.com/AaronErhardt/actix-governor/commit/96626f00b046a99979fb522878f4c6c5d5b3098a
https://github.com/AaronErhardt/actix-governor/commit/3aa7fef8c8a2269a6bc8a9b5217a8e580588756c
https://github.com/AaronErhardt/actix-governor/commit/aece80e074dbdb3e6458c9063c1e6bb37c44bfdd

> 
image

**Comment by TheAwiteb** [comment]
@AaronErhardt There are only these, if they are considered BC https://github.com/AaronErhardt/actix-governor/commit/96626f00b046a99979fb522878f4c6c5d5b3098a

**Comment by AaronErhardt** [comment]
Thanks for checking! No, removing a trait bound is not considered a breaking change, so let's release 0.3.1 :)

**Pull request merged**